pyflakes/test/test_type_annotations.py

```diff
@@ -667,3 +667,22 @@ class C(Protocol):
                 def f():  # type: () -> int
                     pass
         """)
+
+    @skipIf(version_info < (3, 6), 'new in Python 3.6')
+    def test_recursive_global_scope(self):
+        self.flakes("""
+            from typing import Optional, Tuple
+
+            class NestedTuple:
+                x: Tuple[int, Optional['NestedTuple']]
+        """)
+
+    @skipIf(version_info < (3, 6), 'new in Python 3.6')
+    def test_recursive_local_scope(self):
```

Comment:
This used to fail because on the first check the ast just has the "function def" node
So when checking the inside, `self._in_deferred` was now set to `True` causing the issue.
I don't really know why we can't just run everything without it. I remove the  `_in_deferred` boolean and all the unit tests work. I also checked locally on different projects without any issue.
If this causes other problems, I'll be happy to add an extra unit test for this and work on another way to fix it

Reply:
yeah I can't seem to find why I wrote that either -- perhaps I was afraid of modifying the `deferredFunctions` list while it was being iterated -- but it seems append-only-while-iterating works fine in python